internal/qerr/quic_error.go

```diff
@@ -11,6 +11,7 @@ type QuicError struct {
 	ErrorMessage       string
 	isTimeout          bool
 	isApplicationError bool
+	delayed            bool
```

Comment:
It looks like this is only ever true for (1) remote close before handshake completes and (2) version negotiation failure.  I propose we replace "delay" with "isInsecureClose" or similar throughout.  I think that would be clearer.